internal/services/jsonrpc/usersvc/service.go

```diff
@@ -64,26 +75,46 @@ func WithReadTxTimeout(timeout time.Duration) Opt {
 	}
 }
 
-const defaultReadTxTimeout = 5 * time.Second
+func WithAuthenticateRPCs(enabled bool) Opt {
+	return func(cfg *serviceCfg) {
+		cfg.authenticateRPCs = enabled
+	}
+}
+
+func WithChallengeExpiry(expiry time.Duration) Opt {
+	return func(cfg *serviceCfg) {
+		cfg.challengeExpiry = expiry
+	}
+}
+
+const (
+	defaultReadTxTimeout   = 5 * time.Second
+	defaultChallengeExpiry = 10 * time.Second // TODO: or maybe more?
+)
 
 // NewService creates a new instance of the user RPC service.
 func NewService(db DB, engine EngineReader, chainClient BlockchainTransactor,
 	nodeApp NodeApplication, pricer Pricer, migrator Migrator, logger log.Logger, opts ...Opt) *Service {
 	cfg := &serviceCfg{
-		readTxTimeout: defaultReadTxTimeout,
+		readTxTimeout:    defaultReadTxTimeout,
+		challengeExpiry:  defaultChallengeExpiry,
+		authenticateRPCs: false,
 	}
 	for _, opt := range opts {
 		opt(cfg)
 	}
 	return &Service{
-		log:           logger,
-		readTxTimeout: cfg.readTxTimeout,
-		engine:        engine,
-		nodeApp:       nodeApp,
-		pricer:        pricer,
-		chainClient:   chainClient,
-		db:            db,
-		migrator:      migrator,
+		log:              logger,
+		readTxTimeout:    cfg.readTxTimeout,
+		engine:           engine,
+		nodeApp:          nodeApp,
+		pricer:           pricer,
+		chainClient:      chainClient,
+		db:               db,
+		migrator:         migrator,
+		authenticateRPCs: cfg.authenticateRPCs,
+		challengeExpiry:  cfg.challengeExpiry,
```

Comment:
You need to modify `tx-query` because it returns the full transaction body. In private mode, we should make sure to not do this, since it defeats the point.